backend/src/routes/search.ts

```diff
@@ -30,7 +30,10 @@ export async function search(req: Http2ServerRequest, res: Http2ServerResponse):
     const searchService = `https://search-search-api.${mch?.metadata?.namespace || namespace}.svc.cluster.local:4010`
 
     const searchUrl = process.env.SEARCH_API_URL || searchService
-    const endpoint = process.env.globalSearchAPIEndpoint || '/searchapi/graphql'
+    const endpoint =
+      process.env.globalSearchFeatureFlag === 'enabled' && process.env.globalSearchAPIEndpoint
+        ? process.env.globalSearchAPIEndpoint
+        : '/searchapi/graphql'
```

Comment:
Missing the default `/federated` when global search is enabled.

```suggestion
      process.env.globalSearchFeatureFlag === 'enabled' ?
        process.env.globalSearchAPIEndpoint || '/federated'
        : '/searchapi/graphql'
```

Reply:
Maybe this could be simpler and more flexible.

```
const endpoint process.env.searchApiEnpoint || (process.env.globalSearchFeatureFlag === 'enabled' ? '/federated' : /.searchapi/graphql'
```

Note that is changes the name to `searchApiEndpoint` to make it more generic.